toolkit/tools/pkg/imagecustomizerlib/extractpartitions.go

```diff
@@ -190,3 +220,48 @@ func generateRandom128BitNumber() ([SkippableFramePayloadSize]byte, error) {
 	}
 	return randomBytes, nil
 }
+
+// Construct outputPartitionMetadata object.
+func constructOutputPartitionMetadata(diskPartition diskutils.PartitionInfo, partitionNum int, partitionFilepath string, output *[]outputPartitionMetadata) (err error) {
```

Comment:
This code seems far more complicated than it needs to be. Just copy the values from the `PartitionInfo` type to the `outputPartitionMetadata` type.